### DISCUSSION TITLE
Updated release for Custom Environment config?

### DISCUSSION THREAD

**Discussion by charlieforward9** [vitest-dev/vitest#4459] [Q&A] [answered]
I am trying to configure Amplify within my testing environment so I can perform database requests without having to mock data. I have followed the custom environment [doc's](https://vitest.dev/guide/environment.html#custom-environment) and copied the code outlined in the [sample](https://github.com/vitest-dev/vitest/tree/3bec5d1cf568bcba4885d76b05de160212d810fb/test/env-custom). However, I continue to get `MISSING DEP  Can not find dependency 'vitest-environment-./amplify.ts'`

I read into this issue (https://github.com/vitest-dev/vitest/issues/4237), and looked into the associated PR (https://github.com/vitest-dev/vitest/pull/4255), and noticed it was merged into main after 0.34.6 was released. 

I assume this means that these changes are not in v0.34.6, so I attempted to do a direct dependency on the main branch, but then I got issues regarding `Unsupported URL Type "workspace:": workspace:*`. I figure this is not related to the environment configuration, but I am wondering if an updated version of Vitest can be released with (https://github.com/vitest-dev/vitest/pull/4255) included so that the custom environments work.

**Comment by charlieforward9** [comment]
For context, I tried using both a relative path and just the name of the env and both where `MISSING`.

**Comment by sheremet-va** [comment, marked as answer]
You can use vitest beta version (vitest@beta), it should have the fix

**Reply by charlieforward9** [reply]
Hi, thank you for the response. I am realizing that the JSDOM environment I was originally using is essential for my tests to run. Is there a way that I can simply extend the built in JSDOM environment to run my backend config lines within the setup function?

```
    const Amplify = await import("aws-amplify").then((mod) => mod.Amplify);
    const amplifyConfig = await import("./src/amplifyconfiguration.json");
    Amplify.configure(amplifyConfig.default);
```

**Reply by sheremet-va** [reply]
You can do that in a `setupFile`:

```ts
// so it's called before all other imports
await vi.hoisted(async () => {
    const Amplify = await import("aws-amplify").then((mod) => mod.Amplify);
    const amplifyConfig = await import("./src/amplifyconfiguration.json");
    Amplify.configure(amplifyConfig.default);
})
```

**Reply by charlieforward9** [reply]
Thank you for the responses. I have decided to forgo this as it performs network requests and its a little too large of a configuration for now.
